pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -492,7 +510,11 @@ func handleCRDAdd(cl *Client, name string, stop <-chan struct{}) {
 		scope.Errorf("failed to create informer for %v", resourceGVK)
 		return
 	}
-	cl.kinds[s.Resource().GroupVersionKind()] = createCacheHandler(cl, s, i)
+	cl.kinds[resourceGVK] = createCacheHandler(cl, s, i)
+	if w, f := crdWatches[resourceGVK]; f {
```

Comment:
I like the idea of starting various controllers if/when the CRD is detected, we should generalize it.